services/jupyterhub/spawners.py

```diff
@@ -107,15 +108,55 @@ def start(self, *args, **kwargs):
             raise web.HTTPError(401, 'Not authorized to view project.')
             return
 
-        self.image = '{image_registry}'\
-                     '/{namespace}'\
-                     '/{project}'\
-                     ':{commit_sha_7}'.format(
-                        image_registry=os.getenv('IMAGE_REGISTRY'),
-                        commit_sha_7=commit_sha_7,
-                        **options
+        # set default image
+        self.image = os.getenv(
+            'JUPYTERHUB_NOTEBOOK_IMAGE', 'jupyterhub/singleuser:0.8.1'
```

Comment:
IMHO the default is already in the config.

```python
if 'JUPYTERHUB_NOTEBOOK_IMAGE' in os.environ:
    self.image = os.getenv(...)
```

Reply:
isn't that essentially the same as what I wrote?

Reply:
But you are probably experiencing a problem with `self` being global object that is shared by multiple requests.

Reply:
ah that might very well be. I'm not sure how you can reliably avoid this since the base `KubeSpawner` takes `self.singleuser_image_spec` when building the pod manifest. I tried serving multiple requests at once but I agree that this needs a proper test.